### ISSUE TITLE
Bug, on macOS "syntax error near unexpected token `fi'" while running bash script

### ISSUE THREAD

**Issue by maxkatz6** [NetSparkleUpdater/NetSparkle#223]
Full error message
```
: command not founds/0m/8dxrh_r544s_qwl7q1jwjc380000gn/T/ac4f000d-7392-4ed5-80d7-aead450dd478.sh: line 1: 
: command not founds/0m/8dxrh_r544s_qwl7q1jwjc380000gn/T/ac4f000d-7392-4ed5-80d7-aead450dd478.sh: line 2: 
: command not founds/0m/8dxrh_r544s_qwl7q1jwjc380000gn/T/ac4f000d-7392-4ed5-80d7-aead450dd478.sh: line 3: 
/private/var/folders/0m/8dxrh_r544s_qwl7q1jwjc380000gn/T/ac4f000d-7392-4ed5-80d7-aead450dd478.sh: line 10: syntax error near unexpected token `fi'
'private/var/folders/0m/8dxrh_r544s_qwl7q1jwjc380000gn/T/ac4f000d-7392-4ed5-80d7-aead450dd478.sh: line 10: `                            fi;
```

It seems to be related to C# files usually stored with windows-styles line endings "\r\n", while multi line strings are saved with the same line endings as the file in which they are defined, see [stackoverflow (specs)](https://stackoverflow.com/questions/48196840/c-sharp-verbatim-string-line-breaks-crlf-cr-or-lf#comment83374951_48196962). And bash scripts are line-ending sensitive, see [stackoverflow](https://stackoverflow.com/questions/39527571/are-shell-scripts-sensitive-to-encoding-and-line-endings).

**Comment by Deadpikle** [comment]
Hi @maxkatz6, sorry for the wait. I couldn't replicate this only on macOS. Are you building on Windows and then deploying on macOS? I could see how that would cause this issue or maybe having different git settings.

Anyway, can you try the Avalonia build `2.0.7-preview20210808001` or the core `2.0.9-preview20210808001` if you're using the core lib? Thanks, and thanks for researching the data a bit.

If you continue to have issues, can you upload the bash files (and anonymize it if needed)?

**Comment by maxkatz6** [comment]
> Are you building on Windows and then deploying on macOS?

No, building on macOS CI and editing on macOS Rider. But source code file is saved with "\r\n", since most of my team are working on windows with Visual Studio.

Can't check your solution right now, but it's basically the same what I did in my project and it worked (by overriding this method). The only difference that I also added this replace to Exec method for escapedArgs. Not sure if it was required though.
